src/main/java/com/conveyal/r5/analyst/cluster/AnalystWorker.java

```diff
@@ -300,15 +300,15 @@ public void run() {
 
             // Enqueue high-priority (interactive) tasks first to ensure they are enqueued
             // even if the low-priority batch queue blocks.
-            tasks.stream().filter(t -> t instanceof AnalystClusterRequest && ((AnalystClusterRequest) t).outputLocation == null)
+            tasks.stream().filter(AnalystWorker::isHighPriorityTask)
                     .forEach(t -> highPriorityExecutor.execute(() -> {
                         LOG.warn("Handling single point request via normal channel, side channel should open shortly.");
                         this.handleOneRequest(t);
                     }));
 
             // Enqueue low-priority (batch) tasks; note that this may block anywhere in the process
             logQueueStatus();
-            tasks.stream().filter(t -> !(t instanceof AnalystClusterRequest) || ((AnalystClusterRequest) t).outputLocation != null)
+            tasks.stream().filter(t -> !isHighPriorityTask(t))
                 .forEach(t -> {
                     // attempt to enqueue, waiting if the queue is full
                     while (true) {
```

Comment:
This isn't part of the PR but looks pretty questionable to me. The thread sleeps when insertion is rejected? It seems like the executor should handle that kind of thing.